tsl/src/bgw_policy/job.c

```diff
@@ -145,39 +149,108 @@ execute_reorder_policy(BgwJob *job, reorder_func reorder, bool fast_continue)
 	return true;
 }
 
+static bool
+noarg_integer_now_func_filter(Form_pg_proc form, void *arg)
+{
+	Oid *rettype = arg;
+
+	return form->pronargs == 0 && form->prorettype == *rettype;
+}
+
 bool
 execute_drop_chunks_policy(int32 job_id)
 {
 	bool started = false;
 	BgwPolicyDropChunks *args;
+	int64 older_than_datum;
+	Oid table_relid;
+	Oid now_func;
+	Oid rettype;
+	// Q:: could factor out getting dimension type AND now_func oid into a
+	// separate function but not sure where else it would be useful
+	// there is ts_hypertable_get_time_type function that I could use for
+	// time dimension type but it would require a directFunctionCall
+	// and when getting now_func I have access to that info here anyway.
+	Hypertable *hypertable;
+	Cache *hcache;
+	Dimension *open_dim;
 
 	if (!IsTransactionOrTransactionBlock())
 	{
 		started = true;
 		StartTransactionCommand();
+		PushActiveSnapshot(GetTransactionSnapshot());
 	}
 
 	/* Get the arguments from the drop_chunks_policy table */
 	args = ts_bgw_policy_drop_chunks_find_by_job(job_id);
-
+	table_relid = ts_hypertable_id_to_relid(args->fd.hypertable_id);
 	if (args == NULL)
 		ereport(ERROR,
 				(errcode(ERRCODE_TS_INTERNAL_ERROR),
 				 errmsg("could not run drop_chunks policy #%d because no args in policy table",
 						job_id)));
 
-	ts_chunk_do_drop_chunks(ts_hypertable_id_to_relid(args->fd.hypertable_id),
-							IntervalPGetDatum(&args->fd.older_than),
-							0,
-							INTERVALOID,
-							InvalidOid,
-							args->fd.cascade,
-							args->fd.cascade_to_materializations,
-							LOG);
+	if (args->fd.older_than.is_time_interval)
+	{
+		ts_chunk_do_drop_chunks(table_relid,
+								IntervalPGetDatum(&args->fd.older_than.time_interval),
+								(Datum) 0,
+								INTERVALOID,
+								InvalidOid,
+								args->fd.cascade,
+								args->fd.cascade_to_materializations,
+								LOG);
+	}
+	else
+	{
```

Comment:
This branch should be a lot simpler. all the integer - now stuff should be done in interval.c/.h